hierarchy/src/commonMain/kotlin/Hierarchy.kt

```diff
@@ -0,0 +1,22 @@
+package com.juul.krayon.hierarchy
+
+public fun <T> hierarchy(
+    root: T,
+    getChildren: (T) -> List<T>,
+): Node<T, Nothing?> {
+    fun addChildren(parentNode: Node<T, Nothing?>) {
+        val children = getChildren(parentNode.data)
+        parentNode.children = children.map { child ->
+            Node(child, null, parentNode).also { childNode ->
+                addChildren(childNode)
+            }
```

Comment:
nit

```suggestion
            Node(child, null, parentNode)
                .also(::addChildren)
```

_Although this (in the IDE) hides the fact that it is recursive (recursion gutter icon goes away with this syntax)._